examples/multisig_plain/lib.rs

```diff
@@ -457,14 +457,9 @@ mod multisig_plain {
             self.ensure_caller_is_owner();
             let caller = self.env().caller();
             if self.confirmations.take(&(trans_id, caller)).is_some() {
-                if self.confirmation_count.contains_key(&trans_id) {
-                    if let Some(value) = self.confirmation_count.get_mut(&trans_id) {
-                        *value -= 1;
-                    } else {
-                        unreachable!()
-                    }
-                } else {
-                    self.confirmation_count.insert(trans_id, 0);
+                let v_mut = self.confirmation_count.entry(trans_id).or_insert(1);
+                if *v_mut > 0 {
+                    *v_mut -= 1;
```

Comment:
Wouldn't this be the perfect opportunity to make use of https://doc.rust-lang.org/std/collections/btree_map/enum.Entry.html#method.and_modify instead of the `if`?

Reply:
```rust
self.confirmation_count
    .entry(trans_id)
    .and_modify(|v| if v > 0 { *v -= 1; })
    .or_insert(1);
```

Also this would allow us to remove the let binding from what I can see.